test/middleware-test.js

```diff
@@ -0,0 +1,165 @@
+'use strict';
+
+var expect = require('chai').expect;
+var RSVP = require('rsvp');
+var broccoliMiddleware = require('./../lib/middleware');
+var fixture = require('./helpers/fixture-path');
+var TestHTTPServer = require('./helpers/test-http-server');
+
+describe('broccoli-middleware', function() {
+  describe('watcher resolves correctly', function() {
+    var watcher;
+    var server;
+
+    beforeEach(function() {
+      watcher = RSVP.Promise.resolve();
+
+      watcher['builder'] = {};
+    });
+
+    afterEach(function() {
+      server.stop();
+      server = null;
+    })
+
+    it('responds with the given file if file is on disk', function() {
+      watcher['builder']['outputPath'] = fixture('basic-file');
+      var middleware = broccoliMiddleware(watcher, {
+        autoIndex: false
+      });
+
+      server = new TestHTTPServer(middleware);
+
+      return server.start()
+        .then(function(info) {
+          return server.request('/index.html', {
+            info: info
+          });
+        })
+        .then(function(content) {
+          expect(content).to.match(/This is broccoli middleware page/);
+        });
+    });
+
+    it('responds with error if file not found', function(done) {
```

Comment:
@kratiahuja unfortunately this doesn't really work. A test function must not use the `done` callback and return a `Promise` at the same time (see #13).

I tried changing it to the following, but it didn't seem to resolve properly:

```js
    it('responds with error if file not found', function() {
      var deferred = RSVP.defer();

      watcher['builder']['outputPath'] = fixture('basic-file');
      var middleware = broccoliMiddleware(watcher, {
        autoIndex: false
      });

      var wrapperMiddleware = function(req, resp /*next*/) {
        return middleware(req, resp, function(err) {
          expect(resp.finished).to.be.false;
          expect(err.message).to.have.string('ENOENT');
          deferred.resolve();
        })
      };
      server = new TestHTTPServer(wrapperMiddleware);

      return server.start().then(function(info) {
        return server.request('/non-existent-file', {
          info: info
        });
      }).then(function() {
        return deferred.promise;
      });
    });
```